spec/steps/discover.fmf

```diff
@@ -1,44 +1,99 @@
+    Provide method ``tests()`` returning a list of discovered
+    tests and ``requires()`` returning a list of all required
+    packages gathered from individual test metadata.
 
-    * Architectures supported
-    * Disk and memory constraints
-    * Product relevancy
-    * Environment variables
-
-    Examples of metadata storage:
-
-    shell
-        Manual list of shell test cases
-    fmf
-        Flexible Metadata Format filter
-
-    For each ``repository`` the discover step should produce a
-    ``repository.yaml`` file with the list of tests in the
-    following format::
+    Store the list of aggregated tests with their corresponding
+    metadata in the ``tests.yaml`` file. The format should be a
+    dictionary of dictionaries structured in the following way::
 
         /test/one:
-            test: ./test.sh
-            path: /test/path
-            duration: 5m
+            summary: Short test summary.
+            description: Long test description.
+            contact: Petr Šplíchal <psplicha@redhat.com>
+            component: [tmt]
+            test: tmt --help
+            path: /test/path/
+            require: [required, packages]
             environment:
-                X: 1
-                Y: 2
-                Z: 3
+                key1: value1
+                key2: value2
+                key3: value3
+            duration: 5m
+            enabled: true
+            result: respect
+            tag: [tag]
+            tier: 1
+            relevancy: |
+                distro < rhel-8: False
 
         /test/two:
-            test: ./test.sh
-            path: /test/path
-            duration: 10m
+            summary: Short test summary.
+            description: Long test description.
+            ...
+
+/shell:
+    summary: Provide a manual list of shell test cases
+    description:
+        List of test cases to be executed can be defined manually
+        directly in the plan as a list of dictionaries containing
+        test ``name``, actual ``test`` script and optinally a
+        ``path`` to the test.
+    example: |
+        discover:
+            how: shell
+            tests:
+            - name: /help/main
+              test: tmt --help
+            - name: /help/test
+              test: tmt test --help
+            - name: /help/smoke
+              test: ./smoke.sh
+              path: /tests/shell
+    implemented: /tmt/steps/discover/shell
+
+/fmf:
+    summary: Discover available tests using the fmf format
+    description: |
+        Use the `Flexible Metadata Format`_ to explore all
+        available tests in given repository. The following
+        parameters are supported:
+
+        url
+            Git repository containing the metadata tree.
+            Current git repository used by default.
+        ref
+            Branch, tag or commit specifying the desired git
+            revision. Defaults to `master` if url given and the
+            current branch if url not provided.
```

Comment:
Should it be 'HEAD' ? If I'm in detached state, it will just copy 'as-is' when I don't specify the url. Or it will really checkout current branch? Or maybe to say 'will be ignored'?

Reply:
Right you are, it should be current HEAD.